### PR TITLE
Add build performance info to build analysis script 

### PR DESCRIPTION
**Pull request by andygup**
This PR adds performance info to the build analysis script. It runs against all the ESM samples, just like the build-size analysis tool.

The analysis script will now output the following additional information:

 * `elapsedRuntimeMS` - runtime in milliseconds derived from the application's last HTTP request
  * `JSHeapUsedSizeBytes` - JSHeapUsedSize as reported by puppeteer
  * `pageTotalBytes` - total number of bytes calculated using the http response object. 
  * `totalScriptTimeMS` - approximate internal runtime of the library script in milliseconds. Useful for comparing against the `elapsedRuntimeMS`. Should not be used as an indicator of application performance, it's most useful for troubleshooting differences between runtimes and determining if the platform itself caused a slowdown. 

cc @dasa @odoe

### CODE REVIEW

File: .github/scripts/build-perf.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  webserver.start();
+};
+
+/**
+ * Run browser performance tests
+ * @param {string} path
+ * @param {string} sampleInfo
+ * @returns {object} Object containing the performance metrics
+ */
+const browserPerformanceTest = async (path, sampleName = "") => {
```
Comment (benelan) [inline .github/scripts/build-perf.js:126]: What's the reason for setting an empty string as the default for `sampleName`?

File: .github/scripts/build-size.js [hunk 2]
```diff
@@ -106,6 +107,26 @@ const getFileSizeBrotli = (filePath) =>
       help(err, "\n\nOccurred while getting brotli compressed file size.", "Double check the file path:\n   ", filePath)
     );
 
+/**
+ * Provides performance information from an applications's production build
+ * @param {string} buildPath 
+ * @param {string} sampleName 
+ * @returns {object} peformance information
```
Comment (benelan) [inline .github/scripts/build-size.js:114]: pe**r**formance

**Review by benelan — APPROVED** [review]
Looks great! 🚀  Just a couple small doc spelling changes

File: .github/scripts/build-perf.js (hunk 1 above)
Comment (benelan) [reply .github/scripts/build-perf.js:126]: To prevent a null check?

Comment (andygup) [reply .github/scripts/build-perf.js:126]: Yep, prevent a null check

File: .github/scripts/build-size.js [hunk 3]
```diff
@@ -308,9 +332,33 @@ if (require.main === module) {
...
 
       const buildSizes = await getBuildSizes(path, type);
 
+      if(runtime) {
+        const performanceInfo = await getPerformanceInfo(path, buildSizes.mainBundleName);
+
+        // log sizes to console
+        const title = "|> Application Performance <|";
+        const line = "-".repeat(title.length);
+        const bundle = `Main ${type.toUpperCase()} bundle`;
```
Comment (benelan) [inline .github/scripts/build-size.js:345]: One more thing, you could probably delete the line/bundle variables here, and move the ones below up above the runtime conditional. Or move the runtime conditional below the build sizes logging

Comment (andygup) [reply .github/scripts/build-size.js:345]: Ah, good catch.

**Pull request merged**